src/request.rs

```diff
@@ -500,8 +509,10 @@ impl Drop for Request {
 /// Dummy trait that regroups the `Read` and `Write` traits.
 ///
 /// Automatically implemented on all types that implement both `Read` and `Write`.
-pub trait ReadWrite: Read + Write {}
-impl<T> ReadWrite for T where T: Read + Write {}
```

Comment:
i'm 99% sure we don't want to lose this, but it conflicts with existing implementations... not sure how to fix/if this is a code smell?